### PR TITLE
Fix compatability scrbook geometry

### LINKED ISSUE #10: Do not mix `scrbook` and `geometry`
Please don't mix packages `scrbook` and `geometry` without respecting the authors' guidelines. There is a huge amount of typesetting issues coming around when simply `\usepackage{geometry}` in a document based on `\documentclass{scrbook}`.
Please refer to chapter 2 of the [koma script guide](http://mirrors.ctan.org/macros/latex/contrib/koma-script/doc/scrguide-en.pdf) which says

> Many LaTeX classes, including the standard classes, present the user with a largely fixed configuration of margins and page layout. In the standard classes, the choice is limited to selecting a font size. There are separate packages, such as geometry (see [Ume10]), which give the user complete control over, but also full responsibility for, setting the type area and margins.
> KOMA-Script takes a somewhat different approach with the typearea package. Users are offered ways to adjust the design and algorithms based on established typographic standards, making it easier for them to make good choices.

Now of course
> If you have concrete specifications for the margins, typearea is not suitable. In this case, you should use the geometry package (see [Ume10]).

However,
> In the worst case, use the geometry package (see [Ume10]), which is not part of KOMA - Script, or change the page layout parameters of L A TEX yourself. You can find the values as determined by typearea in the log file of your document. The usegeometry option, which you can find in part II, can also improve the interactions between typearea and geometry. This should allow modest adjustments. However, make sure that the proportions of the text area match those of the page, taking the binding correction into account.

### PR DESCRIPTION
**Pull request by philipptempel**
This fixes #10

### CODE REVIEW

File: isw_smb_diss.cls [hunk 1]
```diff
@@ -41,16 +41,31 @@
...
+	headsepline,
+	listof=totoc,% add a tocentry for lists like LOF and LOT
+	toc=chapterentrywithdots,
+	usegeometry,%
+	% DIV=current,%
 ]{scrbook} % KOMA-Script scrreprt, DIN A5, 10pt with bibliography in TOC
 \usepackage[\paperformat]{geometry}
-\geometry{tmargin=3\lenbasemargin,bmargin=3\lenbasemargin,lmargin=2\lenbasemargin,rmargin=2\lenbasemargin,headheight=\lenbasemargin,headsep=\lenbasemargin,footskip=\lenbasemargin,head=\basehead}
+\geometry{%
+	pass,%
```
Comment (chhinze) [inline isw_smb_diss.cls]: Why is this option set? According to <https://texdoc.org/serve/geometry.pdf/0> §5.7 it disables all other geometry options, but no other options are set here. why is this needed?

Comment (philipptempel) [reply isw_smb_diss.cls]: In Chap 2, Markus of `scrbook` explicitly warns (emphasis by me)
> Many LaTeX classes, including the standard classes, present the user with a largely fixed configuration of margins and page layout. In the standard classes, the choice is limited to selecting a font size. There are separate packages, such as  `geometry`  (see [Ume10]), which give the user **complete control** over, but also **full responsibility** for, setting the type area and margins.
> KOMA-Script takes a somewhat different approach with the `typearea` package. **Users are offered ways** to adjust the design and algorithms **based on established typographic standards**, making it easier for them to make good choices.

Further, Markus says that
> As implemented, typesetting rules are usually intended to obtain a useful result even if the author does not know what needs to be considered. Useful frequently means readable and correctable. In my opinion the type area of a text set with LaTeX and the `typearea` package meets these criteria well from the outset. So if you are confronted with regulations which deviate substantially from it, I recommend that you present a sample of the text to your advisor and ask whether you can submit the work despite deviations in the format. If necessary the type area can be adapted somewhat by changing the DIV option. I advise against using `\areaset` for this purpose, however. In the worst case, use the `geometry` package (see [Ume10]), which is not part of KOMA - Script, or change the page layout parameters of LaTeX yourself. You can find the values as determined by `typearea` in the `log` file of your document. The `usegeometry` option, which you can find in part II, can also improve the interactions between `typearea` and `geometry`. This should allow modest adjustments. However, make sure that the proportions of the text area match those of the page, taking the binding correction into account.

Part II of the `scrbook` docs provides information on how to use `geometry` with `scrbook` and `typearea` if the user has to use this combination:
> Usually `typearea` does not care much if you use it with the `geometry` package (see [Ume10]) in any configuration. In particular, this means that `geometry` **does not recognise any changes to the page parameters done with `typearea`**, for example when it changes the paper size — a feature not provided by `geometry` itself.
> Once you set option `usegeometry`, `typearea` tries to translate all of its options into op- tions of geometry. If you activate new parameters inside the document, `typearea` even calls `\newgeometry` (see `\activateareas` in the following section). Since `geometry` does not sup- port changes of paper size or page orientation with `\newgeometry`, `typearea` uses internal commands and `geometry` lengths to provide such changes as needed. This has been tested with `geometry` 5.3 through 5.6.
> Note that using `geometry` and changing page size or orientation with `typearea` does not mean that `geometry` will automatically use the new paper size in an expected manner. For convenience, `geometry` provides far more options to adjust the page than are required to determine the type area, margins, header, footer, etc. — this is called overdetermination — and at the same time `\newgeometry` derives missing information from the known values — known as value preservation — so you often must explicitly specify all new values completely when you call `\newgeometry` on your own. Nevertheless, when `typearea` takes `geometry` into consideration, it opens up additional possibilities.

So, if you use `geometry` with `scrbook` and want to set page margins through `scrbook`, the only way is via package options `usegeometry` for `scrbook` and `pass` for `geometry`.

**Comment by chhinze** [comment]
I'm going to close this as stale. 

However, the main changes are now implemented by passing package options to typearea (not geometry), [see isw_smb_diss.cls, l.53 ff.](https://github.com/iswunistuttgart/isw_smb_diss/blob/9933e24e858710b519881528d40bf46dbe0ef1a8/isw_smb_diss.cls#L53C1-L55C19)

 Credits to @philipptempel for his contribution, thanks a lot.
